modules/twitterstats/client.go

```diff
@@ -0,0 +1,95 @@
+package twitterstats
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// Client contains state that allows stats to be fetched about a list of Twitter users
+type Client struct {
+	httpClient  *http.Client
+	screenNames []string
+}
+
+// TwitterStats Represents a stats snapshot for a single Twitter user at a point in time
+type TwitterStats struct {
+	followerCount int64
+	tweetCount    int64
+}
+
+const (
+	userTimelineURL = "https://api.twitter.com/1.1/users/show.json"
+)
+
+// NewClient creates a new twitterstats client that contains an OAuth2 HTTP client which can be used
+func NewClient(settings *Settings) *Client {
+	usernames := make([]string, len(settings.screenNames))
+	for i, username := range settings.screenNames {
+		switch username.(type) {
+		default:
+			{
+				log.Fatalf("All `screenName`s in twitterstats config must be of type string")
+			}
+		case string:
+			usernames[i] = username.(string)
+		}
```

Comment:
This whole switch can be represented by
```go
if usernames[i], ok = username.(string); !ok {
    log.Fatalf("All `screenName`s in twitterstats config must be of type string")
}
```
(you should declare `var ok bool` before the loop)